src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -376,16 +377,29 @@ private void AddPackageVersionIntoItemGroupCPM(Project project, LibraryDependenc
             directoryBuildPropsRootElement.Save();
         }
 
+        /// <summary>
+        /// Get the Directory package props root element for projects onboarded to CPM.
+        /// </summary>
+        /// <param name="project">Project that needs to be modified.</param>
+        /// <returns>The directory package props root element.</returns>
+        internal ProjectRootElement GetDirectoryPackagePropsRootElement(Project project)
+        {
+            // Get the Directory.Packages.props path.
+            string directoryPackagesPropsPath = project.GetPropertyValue(DirectoryPackagesPropsPathPropertyName);
+            ProjectRootElement directoryPackagePropsRootElement = project.Imports.FirstOrDefault(i => i.ImportedProject.FullPath.Equals(directoryPackagesPropsPath, PathUtility.GetStringComparisonBasedOnOS())).ImportedProject;
```

Comment:
NuGet should not be assuming which file anything is defined in. See our own nuget.client repo as an example, we have a ton of .targets and .props file in the build/ directory, with our repo's build customization.  I fear that this fix might work for customers who put their `GlobalPackageReference` in their Directory.Packages.props, but it will fail for customers who already have it in Directory.Build.props, or in a custom myrepo.props.

Reply:
We should look it up the smae way we lookup PACKAGE_REFERENCE_TYPE_TAG. 

now that I think about it, why are we treating GlobalPackageReference differently from any other packages?

Reply:
These are two great points. We should be aiming for a more seamless integration for MSBuild files. I have updated the code to be not dependent on specific files but rather get the information from the projects

Reply:
You wrote that you updated the code to not dependend on specific files, but this line of code still says `project.Imports.FirstOrDefault(i => i.ImportedProject.FullPath.Equals(directoryPackagesPropsPath, PathUtility.GetStringComparisonBasedOnOS())).ImportedProject`, so it's getting a specific file.

If this PR was about fixing a CPM bug in `dotnet add package`, I could understand this method change. But this PR is about `dotnet list package`, which is read only, so it's not intuitive to me why this `ProjectRootElement GetDirectoryPackagePropsRootElement(Project project)` method should exist at all.

Can you please explain its usage, to help me understand why we need to look up any `ProjectRootElement`?

edit: the only usage appears to be in a method named `AddPackageVersionIntoItemGroupCPM`, which sounds related to `dotnet add package`, not `dotnet list package`. In that case, I'm surprised no tests failed, which suggests we don't have any tests to make sure `dotnet add package` works with CPM. Additionally, we should add a second linked issue that describes the `dotnet add package` bug. Having said that, I really thought `dotnet add package` does already support CPM (writing PackageVersion to Directory.Packages.props)

Reply:
Thanks for pointing that out. The `project.Imports.FirstOrDefault(i => i.ImportedProject.FullPath.Equals(directoryPackagesPropsPath, PathUtility.GetStringComparisonBasedOnOS())).ImportedProject` is a little out of scope for this PR. It is found in a method called `GetDirectoryBuildPropsRootElement` and what it does basically is read the `Directory.Package.Props` file. However the name of the method says otherwise. I was just cleaning up the code so that the name of the method and its actions are in sync. I wil revert these updates since they are out of scope and make the PR confusing.